### ISSUE TITLE
Dropout in skflow has a problem when save & restore the model because the graph collection is not loaded.

### ISSUE THREAD

**Issue by imcomking** [tensorflow/tensorflow#2167]
▶ Operating System: 
ubunutu 14.04 LTS

▶ Installed version of CUDA and cuDNN: 
CUDA 7.5.18 / cuDNN 7.0(v4)
NVIDIA-SMI 352.39     Driver Version: 352.39
GeForce GTX TITAN X

▶ I installed tensorflow with next command line. And that is a link of tensorflow Github, 'Linux GPU, python 2' whl file.

sudo pip install --upgrade http://ci.tensorflow.org/view/Nightly/job/nigntly-matrix-linux-gpu/TF_BUILD_CONTAINER_TYPE=GPU,TF_BUILD_IS_OPT=OPT,TF_BUILD_IS_PIP=PIP,TF_BUILD_PYTHON_VERSION=PYTHON2,label=gpu-working/lastSuccessfulBuild/artifact/pip_test/whl/tensorflow-0.8.0-cp27-none-linux_x86_64.whl

▶ TensorFlow version: 0.8.0
(outputs of python -c "import tensorflow; print(tensorflow.__version__)" )

-----------------------------------------------------------
### Steps to reproduce
1. Save and Restore a tfmodel modifying [tensorflow/tensorflow/examples/skflow/mnist.py] 
![image](https://cloud.githubusercontent.com/assets/4004593/14917371/a93e2882-0e59-11e6-8ff0-8154c0cd318f.png)

2. And see the accuracy is not deterministic but stochastic after restored.
![image](https://cloud.githubusercontent.com/assets/4004593/14917401/d8e70342-0e59-11e6-86f6-aa778b66ede2.png)

### What have you tried?

When I try to predict MNIST data after save() and restore() the 'tfmodel' with skflow function, the dropout shows stochastic outcomes. And when I remove the dropout, there is no stochasticity. 

-----------------------------
I guess when the model has been restored, the collections of a graph about DROPOUT are may not properly loaded. So some codes on
![image](https://cloud.githubusercontent.com/assets/4004593/14917208/dcbbfcc6-0e58-11e6-8b4f-8188a99fda8a.png)
    ▶ feed_dict = {prob: 1.0 for prob in dropouts}
the function of "_predict()" at [tensorflow/contrib/learn/python/learn/estimators/base.py] may dont' work. 

If you run following code after restore tfmodel, then you will see that there is no 'droputs' key.
    ▶ print (classifier._graph.get_all_collection_keys())

**Comment by abronte** [comment]
I'm seeing the same thing. When I remove the dnn I'm not seeing this issue anymore.

**Comment by imcomking** [comment]
@abronte I tried so many things but I couldn't find any solution. Sorry. This is very critical and fundamental problem. Without fixing the graph collection errors, this can't be solved clearly.

**Comment by abronte** [comment]
A work around might when you're done training, save the dropouts object to a pickle 
` dropouts = self._graph.get_collection(DROPOUTS)`

Then when predicting, load that pickle into dropouts?

It looks like a lot of code around this has changed in the master branch, @imcomking have you tried using tensorflow from source to see if this issue still exists?

**Comment by imcomking** [comment]
@abronte No I didn't try yet. I will test the master branch few days later.

**Comment by ilblackdragon** [comment]
Hey guys, indeed we are changing a number of things - this problem should
be fixed though.

That said, currently master doesn't support save / restore for Estimators
the way it was done before. We are working on a bit revised functionality
for this.

On Wed, May 25, 2016 at 12:23 AM, Dong-Hyun Kwak <notifications@github.com>
wrote:

> @abronte <https://github.com/abronte> No I didn't try yet. I will test
> the master branch few days later.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/tensorflow/tensorflow/issues/2167#issuecomment-221493741>
>



-- 
Best regards,
Illia Polosukhin

**Comment by girving** [comment]
@ilblackdragon: If this is fixed at master, can you close it?

**Comment by aselle** [comment]
Automatically closing due to lack of recent activity, please reopen when additional information becomes available.
